extensions/sql-database-projects/src/dialogs/publishDatabaseQuickpick.ts

```diff
@@ -72,15 +73,19 @@ export async function launchPublishDatabaseQuickpick(project: Project): Promise<
 
 	// 2. Select connection
 	const vscodeMssqlApi = await getVscodeMssqlApi();
+	let connectionProfile: IConnectionInfo | undefined = undefined;
+	let connectionUri: string = '';
 	let dbs: string[] | undefined = undefined;
 	while (!dbs) {
-		const connectionProfile = await vscodeMssqlApi.promptForConnection(true);
+		connectionProfile = await vscodeMssqlApi.promptForConnection(true);
 		if (!connectionProfile) {
+			// User cancelled
 			return;
 		}
 		// Get the list of databases now to validate that the connection is valid and re-prompt them if it isn't
 		try {
-			dbs = await vscodeMssqlApi.listDatabases(connectionProfile);
+			connectionUri = await vscodeMssqlApi.connect(connectionProfile);
+			dbs = await vscodeMssqlApi.listDatabases(connectionUri);
```

Comment:
do you remember if this issue https://github.com/microsoft/azuredatastudio/issues/9361 for `listDatabases()` failing if the user doesn't have access to master got fixed? I saw that the PR to fix it in sqltoolsservice got reverted.

Reply:
Not sure, but that shouldn't be an issue that's only for the vs code extension right? We call listDatabases in ADS as well to populate the DB list so if that was broken it would affect both scenarios.